src/components/buttonSelectors/MultiButtonSelector.js

```diff
@@ -1,117 +1,159 @@
-            updateResponse && updateResponse({id, question: currId, selection: idCode})
-        }else{
-            let vehicleID = dependencyID[1]
-            switch (dependencyID.length) {
-                case 2:
-                    updateResponse && updateResponse({id, question: currId, selection: idCode, vehicleID: vehicleID})
-                    break;
-                case 3:
-                    let passengerID = dependencyID[2]
-                    updateResponse && updateResponse({id, question: currId, selection: idCode, vehicleID: vehicleID, passengerID: passengerID})
-                case 4:
-                    updateResponse && updateResponse({id, question: currId, selection: idCode, vehicleID: vehicleID, nonmotoristID: dependencyID[3]})
-                default:
-                    break;
-            }
-        }
-        submitFunction({id, question: currId, selection: optionText})
+  /*
+    This function saves the value of the  selected button to the form reducer.
+  */
+  const submitField = (optionText, idCode) => {
+    setSelection(optionText);
+    if (dependencyID == null || dependencyID.length == 1) {
+      updateResponse &&
+        updateResponse({ id, question: currId, selection: idCode });
+    } else {
+      let vehicleID = dependencyID[1];
+      switch (dependencyID.length) {
+        case 2:
+          // Case 2 updates the vehicle form
+          updateResponse &&
+            updateResponse({
+              id,
+              question: currId,
+              selection: idCode,
+              vehicleID: vehicleID,
+            });
+          break;
+        case 3:
+          // Case 3 updates the passenger form
+          let passengerID = dependencyID[2];
+          updateResponse &&
+            updateResponse({
+              id,
+              question: currId,
+              selection: idCode,
+              vehicleID: vehicleID,
+              passengerID: passengerID,
+            });
+        case 4:
+          // Case 4 updates the nonmotorist form
+          updateResponse &&
+            updateResponse({
+              id,
+              question: currId,
+              selection: idCode,
+              vehicleID: vehicleID,
+              nonmotoristID: dependencyID[3],
+            });
+        default:
+          break;
+      }
     }
+    submitFunction({ id, question: currId, selection: optionText });
+  };
 
-    const renderSingleButton = (option) => {
-        return (
-            <Button
-                key={option.id}
-                mr={4}
-                mb={4}
-                variant={selection === option.name ? 'solid': 'subtle'}
-                onPress={() => submitField(option.name, option.id)}
-            >
-                {option.name}
-            </Button>
-        )
-    }
+  /*
+    This function renders each of the buttons and sets their styling.
+  */
+  const renderSingleButton = (option) => {
+    return (
+      <Button
+        key={option.id}
```

Comment:
just noting that we may want to create an issue about the repeated key warning - possibly suggest changing the key to a uuid or the option name instead